bugfix: S3C-2201: econnreset rest client keep alive

alexandre-merle opened the pull request:
Use agentkeepalive to avoid econnreset on client sockets, more info
in S3C-3114.

Fixes https://scality.atlassian.net/browse/S3C-2201

bert-e commented:
# Hello alexandre-merle,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/arsenal/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/7.9
 $ git merge origin/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

alexandre-merle commented:
ping

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/8.1
 $ git merge origin/w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

alexandre-merle commented:
ping

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-2201-econnreset-rest-client-keep-alive` into
`development/7.4`
* [w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive](https://github.com/scality/arsenal/compare/development/7.9...w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive) will be merged into `development/7.9` (pull request #1335)
* [w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive) will be merged into `development/8.1` (pull request #1336)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

naren-rajendran commented on lib/network/rest/RESTClient.js:
```diff
@@ -8,6 +8,13 @@ const constants = require('../../constants');
+// rest server is configured to wait 60 seconds of inactivity on a
+// HTTP socket before closing it, so we might reuse it safely for a
+// little bit less time to avoid ECONNRESET issues
+const FREE_SOCKET_TIMEOUT = 55000;
```
should we reuse https://github.com/scality/Arsenal/blob/development/7.4/lib/constants.js#L85 ?

alexandre-merle replied:
nice catch, didn't know it was here, but im more concern by the fact that this do not fix the issue for good

naren-rajendran replied:
Can we reuse the constant from constants? :)

alexandre-merle replied:
yes i will, but i put this on hold the time i investigate why this fix is not enought :)

alexandre-merle replied:
done

naren-rajendran commented on lib/network/rest/RESTServer.js:
```diff
@@ -73,6 +73,12 @@ function parseURL(urlStr, expectKey) {
+// timeout (55 seconds) to avoid the risk of triggering ECONNRESET
+// errors if the server closes the connection at the time rest client
+// attempts to reuse it.
+const KEEP_ALIVE_TIMEOUT = 60000;
```
should we reuse https://github.com/scality/Arsenal/blob/development/7.4/lib/constants.js#L85 ?

alexandre-merle replied:
done

bert-e commented:
# History mismatch



Merge commit #19f7ab18f94200381ce73da3ee664ae69392e8ca on the integration branch
`w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive` is merging a branch which is neither the current
branch `bugfix/S3C-2201-econnreset-rest-client-keep-alive` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-2201-econnreset-rest-client-keep-alive` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-merle commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/7.9
 $ git merge origin/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/8.1
 $ git merge origin/w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

alexandre-merle committed:
bugfix: S3C-2201: econnreset rest client keep alive

Use agentkeepalive to avoid econnreset on client sockets, more info
in S3C-3114.

Fixes https://scality.atlassian.net/browse/S3C-2201

bert-e commented:
# History mismatch



Merge commit #6bd61af2481205439d060b0a389a887ec6bf59fb on the integration branch
`w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive` is merging a branch which is neither the current
branch `bugfix/S3C-2201-econnreset-rest-client-keep-alive` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-2201-econnreset-rest-client-keep-alive` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-merle commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/7.9
 $ git merge origin/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive` with contents from `w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive origin/development/8.1
 $ git merge origin/w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive
```

alexandre-merle commented:
ping

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-2201-econnreset-rest-client-keep-alive` into
`development/7.4`
* [w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive](https://github.com/scality/arsenal/compare/development/7.9...w/7.9/bugfix/S3C-2201-econnreset-rest-client-keep-alive) will be merged into `development/7.9` (pull request #1337)
* [w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/bugfix/S3C-2201-econnreset-rest-client-keep-alive) will be merged into `development/8.1` (pull request #1338)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

alexandre-merle commented:
@ironman-machine still there body ?

alexandre-merle commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`



Please check the status of the associated issue S3C-2201.

Goodbye alexandre-merle.
